### PR TITLE
Make supporters searchable by phone number

### PR DESCRIPTION
**Pull request by wwahammy**
Doesn't have payments searchable by phone number yet.

Closes https://github.com/CommitChange/tix/issues/3192

### CODE REVIEW

File: lib/query/query_supporters.rb [hunk 1]
```diff
@@ -231,6 +231,11 @@ def self.full_filter_expr(np_id, query)
     if query[:search].present?
       expr = expr.and_where(%Q(
         supporters.fts @@ websearch_to_tsquery('english', $search)
+        OR (
+          supporters.phone IS NOT NULL
+          AND supporters.phone != ''
+          AND supporters.phone_index = (regexp_replace($search, '\\D','', 'g'))
+        )
```
Comment (clarissalimab) [inline lib/query/query_supporters.rb]: I think it would be good to add tests for this, what do you think?

**Review by clarissalimab — COMMENTED** [review]
:speech_balloon::

File: lib/query/query_supporters.rb (hunk 1 above)
Comment (wwahammy) [reply lib/query/query_supporters.rb]: That's a good idea!

**Review by clarissalimab — APPROVED** [review]
:rocket: nice!

**Pull request merged**